### PR TITLE
feat: Add example code snippets for fine grained access control tokens

### PR DESCRIPTION
**Pull request by krispraws**
- Add code examples of generating tokens with different permissions for GenerateAuthToken docs
-  ~Builds will fail on the PR workflow jobs because the SDK changes are not yet merged/released but I have run these locally.~ Fixed by updating sdk version after new momento sdk was released

```
> momento-nodejs-example@1.0.0 doc-examples
> tsc && node dist/doc-example-files/doc-examples-js-apis.js

[2023-07-11T01:00:29.152Z] INFO (Momento: CacheClient): Creating Momento CacheClient
[2023-07-11T01:00:29.170Z] INFO (Momento: CacheClient): Creating Momento CacheClient
[2023-07-11T01:00:29.171Z] INFO (Momento: ControlClient): Creating cache: test-cache
Cache 'test-cache' created
Key 'test-key' was not found in cache 'test-cache'
Key 'test-key' stored successfully
[2023-07-11T01:00:29.686Z] INFO (Momento: ControlClient): Deleting cache: test-cache
Cache 'test-cache' deleted
[2023-07-11T01:00:29.713Z] INFO (Momento: ControlClient): Creating cache: test-cache
Cache 'test-cache' created
Caches:
default-cache
momento-loadgen
test-cache
cache


Cache 'test-cache' flushed
Key 'test-key' stored successfully
Retrieved value for key 'test-key': test-value
Key 'test-key' deleted successfully
Key 'test-key' incremented successfully. New value in key test-key: 11
Item type retrieved successfully: SCALAR
Key 'test-key' already exists in cache 'test-cache', so we did not overwrite it
List fetched successfully: a,b,c
Values added successfully to the back of the list 'test-list'. Result - Success: listLength: 9
Values added successfully to the front of the list 'test-list'. Result - Success: listLength: 15
Length of list 'test-list' is 18
Last value was removed successfully from list 'test-list': three
First value was removed successfully from list 'test-list': one
Value 'x' added successfully to back of list 'test-list'
Value 'x' added successfully to front of list 'test-list'
Value 'b' removed successfully from list 'test-list'
Retaining elements from index 1 to 4 from list 'test-list'
Dictionary fetched successfully-
test-field : test-value
Field test-field fetched successfully from cache 'test-cache' with value: test-value
Values fetched successfully-
key1 : value1
key2 : value2
Field set successfully into cache 'test-cache'
Fields set successfully into cache 'test-cache'
Field value incremented by 1. Result - 11
Field removed successfully from dictionary 'test-dictionary'
Fields removed successfully from dictionary 'test-dictionary'
Element added successfully to set 'test-set'
Elements added successfully to set 'test-set'
Set fetched successfully-
test-element1 : test-element1
test-element2 : test-element2
test-element : test-element
Element 'test-element' removed successfully from set 'test-set'
Elements 'test-element1' and 'test-element2' removed successfully from set 'test-set'
Value 'test-value' with score '5' added successfully to sorted set 'test-sorted-set'
Elements added successfully to sorted set 'test-sorted-set'
Values from sorted set 'test-sorted-set' fetched by rank successfully-
test-value : 5
key1 : 10
key2 : 20
Values from sorted set 'test-sorted-set' fetched by score successfully-
test-value : 5
key2 : 25
key1 : 100
Element with value 'key1' has rank: 2
Element with value 'key1' has score: 10
Element scores retrieved successfully -
key1 : 10
key2 : 20
Score for value 'test-value' incremented successfully. New score - 11
Element with value 'test-value' removed successfully
Elements with value 'key1' and 'key2 removed successfully
Generated an auth token with AllDataReadWrite scope!
Auth token starts with: eyJlbmRwb2
Refresh token starts with: eyJhbGciOi
Expires At: 1689039031
Generated an auth token with read-only access to cache foo!
Auth token starts with: eyJlbmRwb2
Refresh token starts with: eyJhbGciOi
Expires At: 1689039031
Generated an auth token with read-write access to all caches!
Auth token starts with: eyJlbmRwb2
Refresh token starts with: eyJhbGciOi
Expires At: 1689039031
Generated an auth token with publish-subscribe access to all topics within cache bar!
Auth token starts with: eyJlbmRwb2
Refresh token starts with: eyJhbGciOi
Expires At: 1689039031
Generated an auth token with subscribe-only access to topic where_is_mo within cache mo_nuts!
Auth token starts with: eyJlbmRwb2
Refresh token starts with: eyJhbGciOi
Expires At: 1689039031
Generated an auth token with multiple cache and topic permissions!
Auth token starts with: eyJlbmRwb2
Refresh token starts with: eyJhbGciOi
Expires At: 1689039032
Generated auth token; refreshing!
Auth token refreshed!
Refreshed auth token starts with: eyJlbmRwb2
New refresh token starts with: eyJhbGciOi
Refreshed auth token expires At: 1689039032
[2023-07-11T01:00:32.233Z] INFO (Momento: TopicClient): Creating Momento CacheClient
[2023-07-11T01:00:32.237Z] INFO (Momento: TopicClient): Creating Momento CacheClient
Value published to topic 'test-topic'
Successfully subscribed to topic 'test-topic'
Publishing values to the topic 'test-topic': test-value
Waiting for the published values
```

### CODE REVIEW

File: examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts [hunk 1]
```diff
@@ -781,13 +786,93 @@ function example_API_InstantiateTopicClient() {
...
+  if (oneCacheOneTopicRWTokenResponse instanceof GenerateAuthToken.Success) {
+    console.log('Generated an auth token with subscribe-only access to topic where_is_mo within cache mo_nuts!');
+    // logging only a substring of the tokens, because logging security credentials is not advisable :)
+    console.log(`Auth token starts with: ${oneCacheOneTopicRWTokenResponse.authToken.substring(0, 10)}`);
+    console.log(`Refresh token starts with: ${oneCacheOneTopicRWTokenResponse.refreshToken.substring(0, 10)}`);
+    console.log(`Expires At: ${oneCacheOneTopicRWTokenResponse.expiresAt.epoch()}`);
+  }
+
+  // Generate a token with multiple permissions
+  const permissions = {
```
Comment (NoSQLKnowHow) [inline examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts:852]: Is it possible to split this out to be two different variables? One for TopicsPermissions and one for CachePermissions? and add more comments like here https://github.com/momentohq/public-dev-docs/blob/chore/new-tokens-docs/docs/develop/api-reference/auth-tokens.md#permission-object-examples

Comment (NoSQLKnowHow) [reply examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts:852]: Yep, my bad. You're correct. I have it labeled wrong.

What I am trying to show are examples of all 4 of the managed roles and the use of either AllCaches or strings passed in. So it is brutally obvious what the possibilities are.

Comment (NoSQLKnowHow) [reply examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts:852]: I pushed a version labelled correctly.

**Review by mjindalmotiv8 — DISMISSED** [review]
Looks good - as customers ask questions in discord we can keep on adding but this list looks good

**Review by malandis — COMMENTED** [review]
Can you demonstrate the common errors one would see? Eg cache does not exist, invalid permission combination, ...

**Comment by krispraws** [comment]
> Can you demonstrate the common errors one would see? Eg cache does not exist, invalid permission combination, ...

Caches and topics do not need to exist when the token is created. Invalid permission combination would make it very verbose. I will add a check for error in the snippet

**Comment by malandis** [comment]
> Caches and topics do not need to exist when the token is created. Invalid permission combination would make it very verbose. I will add a check for error in the snippet

👍  That's fine -- I meant to just add error handling for the common failure cases.

**Review by bruuuuuuuce — DISMISSED** [review]
:shipit:

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Review by malandis — APPROVED** [review]

**Pull request merged**